audio_program_generator/apg.py

```diff
@@ -99,43 +108,30 @@ def gen_speech(self):
         """Generate a combined speech file, made up of gTTS-generated speech
         snippets from each line in the phrase_file + corresponding silence."""
 
-        with open(self.phrase_file, "r") as f:
-            #pbar = ProgressBar(maxval=num_lines_in_file(self.phrase_file)).start()
-            combined = AudioSegment.empty()
-            lines = f.readlines()
-
-            for line in tqdm(lines):
-                try:
-                    phrase, interval = line.split(";")
-                except Exception as e:
-                    print("Error parsing input file as CSV:")
-                    print(line)
-                    print(e.args)
-                    sys.exit()
-
-                if len(phrase) == 0:
-                    print("Error: gTTS requires non-empty text to process.")
-                    print("File: ", self.phrase_file)
-                    print("Line: ", line)
-                    sys.exit()
-
-                # Each speech snippet generated from gTTS is saved locally
-                # in a cache (if that exact phrase had not already been used).
-                # Otherwise, it is simply re-used.
-                Path.mkdir(Path.cwd() / ".cache") if not Path(
-                    Path.cwd() / ".cache"
-                ).exists() else None
-                file = Path.cwd() / ".cache" / (phrase + ".mp3")
-                if not Path(file).exists():
-                    speech = gTTS(phrase)
-                    speech.save(file)
-
-                # Add the current speech snippet + corresponding silence
-                # to the combined file, building up for each new line.
-                speech = AudioSegment.from_file(file, format="mp3")
-                combined += speech
-                silence = AudioSegment.silent(duration=1000 * int(interval))
-                combined += silence
+        combined = AudioSegment.empty()
+
+        for phrase, duration in tqdm(parse_textfile(self.phrase_file)):
+
+            # gTTS throws exception if given nothin-string
+            if phrase.strip() == "":
+                continue
+
+            # Cache genearted gTTTS snippets and reuse if already present
+            Path.mkdir(Path.cwd() / ".cache") if not Path(
+                Path.cwd() / ".cache"
+            ).exists() else None
+
+            file = Path.cwd() / ".cache" / (phrase + ".mp3")
```

Comment:
f"{phrase}.mp3", and with above you can reuse `cache_dir` variable here

Reply:
Will implement when addressing [cache enhancement](https://github.com/jeffwright13/audio_program_generator/issues/11).